request_information.go

```diff
@@ -498,14 +498,17 @@ func (request *RequestInformation) AddQueryParameters(source any) {
 		str, ok := value.(*string)
 		if ok && str != nil {
 			request.QueryParameters[fieldName] = *str
+			continue

```

Comment:
this will introduce a regression skipping normalization below.

updating to 

```go
		if str, ok := value.(*string); ok && str != nil {
```
 and else if 
and removing the continues
should achieve the desired performance improvement